From 7191fb1074ba4108e56fb48a69dd1a30a67d7b41 Mon Sep 17 00:00:00 2001
From: Felix Radensky <felix.r@variscite.com>
Date: Mon, 4 May 2020 11:49:51 +0300
Subject: [PATCH 3/4] imx8mn: Allocate UART2 to M7 domain only if M7 is enabled

i.MX8MN ATF code unconditionally allocates UART2 and UART4 to M7 domain.
On Variscite i.MX8MN board UART4 is used by Linux running on A53 domain.
Allocate only UART2 to M7 domain and only if M7 is enabled.
---
 plat/imx/imx8mn/imx8mn_bl31_setup.c | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/plat/imx/imx8mn/imx8mn_bl31_setup.c b/plat/imx/imx8mn/imx8mn_bl31_setup.c
index 57db28a..68fd169 100644
--- a/plat/imx/imx8mn/imx8mn_bl31_setup.c
+++ b/plat/imx/imx8mn/imx8mn_bl31_setup.c
@@ -242,10 +242,11 @@ void bl31_early_platform_setup2(u_register_t arg0, u_register_t arg1,
 #endif
 	bl31_tzc380_setup();
 
-	/* Assign M7 to domain 1 */
-	mmio_write_32(IMX_RDC_BASE + 0x204, 0x1);
-	mmio_write_32(IMX_RDC_BASE + 0x518, 0xfc);
-	mmio_write_32(IMX_RDC_BASE + 0x5A4, 0xf3);
+	/* Assign UART2 to domain 1 */
+	if (imx_is_m4_enabled()) {
+		mmio_write_32(IMX_RDC_BASE + 0x204, 0x1);
+		mmio_write_32(IMX_RDC_BASE + 0x5A4, 0xf3);
+	}
 }
 
 void bl31_plat_arch_setup(void)
-- 
2.7.4

